From 5775a83685d27112b60261d6bf6322ddc976651d Mon Sep 17 00:00:00 2001
From: Xinyu Chen <xinyu.chen@freescale.com>
Date: Tue, 19 Apr 2011 15:35:56 +0800
Subject: [PATCH 137/635] ENGR00142312 libgralloc: init ipu lib IPC in android
 fb HAL

CTS test forbid the write permission for other group to shared memory
directory: /mnt/shm, so we must create the IPU semaphore share files
on boot when framebuffer mapped by the system server who's id is system:system.

Signed-off-by: Xinyu Chen <xinyu.chen@freescale.com>
---
 libgralloc/framebuffer.cpp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/libgralloc/framebuffer.cpp b/libgralloc/framebuffer.cpp
index a7e3df1..0e170df 100755
--- a/libgralloc/framebuffer.cpp
+++ b/libgralloc/framebuffer.cpp
@@ -42,11 +42,9 @@
 #include <linux/videodev.h>
 #include <sys/mman.h>
 
-#ifdef SECOND_DISPLAY_SUPPORT
 extern "C" {
 #include "mxc_ipu_hl_lib.h" 
 } 
-#endif
 
 #endif
 #include <GLES/gl.h>
@@ -1355,6 +1353,11 @@ int fb_device_open(hw_module_t const* module, const char* name,
             const_cast<int&>(dev->device.maxSwapInterval) = 1;
             *device = &dev->device.common;
         }
+
+#ifndef FSL_EPDC_FB
+	/* initialize the IPU lib IPC */
+	mxc_ipu_lib_ipc_init();
+#endif
     }
     return status;
 }
-- 
1.8.0

